src/local.jl

```diff
@@ -156,36 +159,20 @@ function _path_tz_name(path::AbstractString, mask::Class=Class(:ALL))
 end
 
 
-# Using conditional expressions `(?(condition)yes-regexp)` as control flow to indicate that
-# that a captured group is dependent on a previous group begin matched. This could also be
-# accomplished using nested groups.
-const TZ_REGEX = r"""
-    ^
-    (?<name>[a-zA-Z]{3,})?
-    (?(name)(?<sign>[+-]))?
-    (?(name)(?<hour>\d+))?
-    (?(hour)\:(?<minute>\d+))?
-    (?(minute)\:(?<second>\d+))?
-    $
-    """x
-
 """
     parse_tz_format(str) -> TimeZone
 
 Parse the time zone format typically provided via the "TZ" environment variable. Details on
 the format can be found under the man page for
 [tzset](http://man7.org/linux/man-pages/man3/tzset.3.html).
-
-Currently this function handles only the first format which is a fixed time zone without
-daylight saving time.
 """
 function parse_tz_format(str::AbstractString)
-    tz = tryparse_tz_format(str)
-
-    if tz !== nothing
+    x = _parsesub_tz(str)
+    if x isa Tuple
+        tz, i = x
         return tz
```

Comment:
```suggestion
    if x isa Tuple
        return x[1]
```

We're also duplicating this code below on [here](https://github.com/JuliaTime/TimeZones.jl/pull/259/files#diff-8044bd4fb98ddd8eb84ea707c8d7d00bR186), maybe make this into its own function?

Reply:
I prefer the way it was implemented as having the explicit `tz` variable name makes it clearer what is being returned. As for the duplicate code I'm still iterating on this interface. Any suggestions for what this function should look like?

Reply:
Fair enough. Alternatively we could do something similar to below (note; `_try` is quite a poor name);

```julia
function parse_tz_format(str::AbstractString; _try=false)
    x = _parsesub_tz(str)
    if x isa Tuple
        tz, i = x
        return tz
    else
        return _try ? nothing : throw(ParseNextError("Unhandled TZ environment variable. $(x.msg)", x.str, x.s, x.e))
    end
end
```

Reply:
I should point out that `parse` and `tryparse` are conventions set by Julia. I'll note the `parsesub` idea is my attempt to improve on the design.

I could combine these functions but I think it would need to be something like:
```julia
for f in (:parse, :tryparse)
    @eval function $(f)_tz_format(...)
        ...
    end
end
```

Reply:
Ah didn't know this was a standard practice in Julia. In that case this can stay as is, we're not really gaining anything from combining them in this case and it may just be complicated to read.